cfme/tests/infrastructure/test_provisioning.py

```diff
@@ -204,5 +231,29 @@ def verify():
                 in normalize_text(mail["subject"]),
                 smtp_test.get_emails())) == len(vm_names)
         )
-
     wait_for(verify, message="email receive check", delay=5)
+
+
+@pytest.mark.parametrize("approve", [True, False], ids=["approved", "denied"])
+def test_provision_approval_msg(
+        setup_provider, provider, vm_name, request, approve, provision_request):
+    template, prov_data = provision_request
+    prov_data['num_vms'] = "2"
+    if approve:
+        do_vm_provisioning(template, provider, vm_name, prov_data, request, smtp_test=False,
+                       wait=False)
+    else:
+        do_vm_provisioning(template, provider, vm_name, prov_data, request, smtp_test=False,
+                       wait=False)
+    cells = {'Description': 'Provision from [{}] to [{}###]'.format(template, vm_name)}
+    if approve:
+        requests.approve_request(cells, "Approved")
+        request.addfinalizer(lambda: requests.delete_request(cells))
+    else:
+        requests.deny_request(cells, "Denied")
+    row, __ = wait_for(requests.find_request, [cells],
+                       fail_func=requests.reload, num_sec=1500, delay=20)
+    if approve:
+        assert normalize_text(row.reason.text) == 'approved'
+    else:
+        assert normalize_text(row.reason.text) == 'denied'
```

Comment:
honestly, from my perspective it would be better to make two tests of this one. There are too many if/elses for both cases.